homeassistant/components/digital_ocean/__init__.py

```diff
@@ -81,6 +89,31 @@ def get_droplet_id(self, droplet_name):
 
         return droplet_id
 
+    @Throttle(MIN_TIME_BETWEEN_DOMAIN_UPDATES)
+    def update_domain_record(
+        self, domain_name, record_name, record_value, record_type="A"
+    ):
+        """Update the appointed DNS record with the desired value."""
+        domain = digitalocean.Domain(token=self._access_token, name=domain_name)
```

Comment:
What happens when the access token expires ?  Do you need to handle that here?

Reply:
Well, obviously the call will fail. But please note that this is a pre-existing problem.
My code merely touches on re-using the existing integration & token already set up to update DNS records.

Ideally, I would like to add configuration flow to this PR, but:
1. This was my first contribution, so I started slow and wanted to get some feedback before dedicating too much time to it.
2. It self kinda out of the scope of this PR. I'm open to adding it to another PR/feature if you deem important.

Reply:
@austinmroczek Please advise.

Reply:
Waiting for a config flow PR makes sense

Reply:
Nice! Thank you!